app/core/event_handler.py

```diff
@@ -13,38 +13,40 @@
 async def handle_event(premis_event: PremisEvent) -> None:
     log.debug(
         "Start handling of PREMIS event.",
-        fragment_id=premis_event.fragment_id,
+        mediahaven_id=premis_event.mediahaven_id,
     )
 
     mediahaven_service = MediahavenService(config.app_cfg)
-    fragment_id = premis_event.fragment_id
+    mediahaven_id = premis_event.mediahaven_id
 
     # Get metadata for the newly archived item
     try:
-        fragment = mediahaven_service.get_fragment(fragment_id)
-        original_pid = fragment["Dynamic"]["s3_object_key"].split(".")[0]
+        fragment = mediahaven_service.get_fragment(mediahaven_id)
     except MediaObjectNotFoundException as e:
         log.critical(
```

Comment:
```suggestion
        log.error(
```
Here as well: it is a wildly strange exception but the application can handle this exception just fine. ;-)